api/order_test.go

```diff
@@ -16,6 +17,182 @@ import (
 )
 
 // ------------------------------------------------------------------------------------------------
+// CREATE
+// ------------------------------------------------------------------------------------------------
+func TestOrderCreationWithSimpleOrder(t *testing.T) {
+	db, config := db(t)
+	config.LogConf.Level = "DEBUG"
+	ctx := testContext(nil, config, false)
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case "/test-product":
+			fmt.Fprintln(w, `<!doctype html>
+				<html>
+				<head><title>Test Product</title></head>
+				<body>
+					<script class="netlify-commerce-product">
+					{"sku": "product-1", "title": "Product 1", "type": "Book", "prices": [
+						{"amount": "9.99", "currency": "USD"}
+					]}
+					</script>
+				</body>
+				</html>`)
+		case "/netlify-commerce/settings.json":
+			fmt.Fprintln(w, `{
+			  "taxes": [
+					{"percentage": 19, "product_types": ["E-Book"], "countries": ["Germany"]},
+					{"percentage": 7, "product_types": ["Book"], "countries": ["Germany"]}
+				]
+			}`)
+		}
+	}))
+
+	config.SiteURL = ts.URL
+
+	recorder := httptest.NewRecorder()
+	req, _ := http.NewRequest("POST", "https://not-real", strings.NewReader(`{
+		"email": "info@example.com",
+		"shipping_address": {
+			"first_name": "Test", "last_name": "User",
+			"address1": "610 22nd Street",
+			"city": "San Francisco", "state": "CA", "country": "USA", "zip": "94107"
+		},
+		"line_items": [{"path": "/test-product", "quantity": 1}]
+	}`))
+	api := NewAPI(config, db, nil, nil)
+
+	api.OrderCreate(ctx, recorder, req)
+
+	order := &models.Order{}
+	extractPayload(t, 201, recorder, order)
+
+	var total uint64
+	total = 999
+	assert.Equal(t, "info@example.com", order.Email, "Total should be info@example.com, was %v", order.Email)
+	assert.Equal(t, total, order.Total, fmt.Sprintf("Total should be 999, was %v", order.Total))
+}
+
+func TestOrderCreationWithTaxes(t *testing.T) {
+	db, config := db(t)
+	ctx := testContext(nil, config, false)
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case "/test-product":
+			fmt.Fprintln(w, `<!doctype html>
+				<html>
+				<head><title>Test Product</title></head>
+				<body>
+					<script class="netlify-commerce-product">
+					{"sku": "product-1", "title": "Product 1", "type": "Book", "prices": [
+						{"amount": "9.99", "currency": "USD"}
+					]}
+					</script>
+				</body>
+				</html>`)
+		case "/netlify-commerce/settings.json":
+			fmt.Fprintln(w, `{
+			  "taxes": [
+					{"percentage": 19, "product_types": ["E-Book"], "countries": ["Germany"]},
+					{"percentage": 7, "product_types": ["Book"], "countries": ["Germany"]}
+				]
+			}`)
+		}
+	}))
+
+	config.SiteURL = ts.URL
+
+	recorder := httptest.NewRecorder()
+	req, _ := http.NewRequest("POST", "https://not-real", strings.NewReader(`{
```

Comment:
You could also take this out to a helper method. And instead of doing a string literal I will usually build a struct and serialize it. That isn't necessary though